deepfence_server/handler/scan_reports.go

```diff
@@ -1822,49 +1820,96 @@ func extractBulksNodes(nodes []model.NodeIdentifier) (regularNodes []model.NodeI
 	return regularNodes, clusterNodes, registryNodes
 }
 
+func IsScanInProgress(ctx context.Context, scan_type utils.Neo4jScanType, node model.NodeIdentifier) bool {
+	driver, err := directory.Neo4jClient(ctx)
+	if err != nil {
+		return false
+	}
+
+	session := driver.NewSession(neo4j.SessionConfig{AccessMode: neo4j.AccessModeRead})
+	if err != nil {
+		return false
+	}
+	defer session.Close()
+
+	tx, err := session.BeginTransaction(neo4j.WithTxTimeout(30 * time.Second))
+	if err != nil {
+		return false
+	}
+	defer tx.Close()
+
+	query := fmt.Sprintf(`Match (n:%s{node_id:$node_id}) return n.%s as status`,
+		utils.NEO4J_NODE_TYPE[node.NodeType], utils.ScanStatusField[scan_type])
+
+	resp, err := tx.Run(query, map[string]interface{}{"node_id": node.NodeId})
+	if err != nil {
+		return false
+	}
+
+	record, err := resp.Single()
+	if err != nil {
+		return false
+	}
+
+	val, found := record.Get("status")
+	if !found {
+		return false
+	}
+
+	status := val.(string)
+	if status == utils.SCAN_STATUS_STARTING || status == utils.SCAN_STATUS_INPROGRESS ||
+		status == utils.SCAN_STATUS_CANCELLING || status == utils.SCAN_STATUS_CANCEL_PENDING {
+		return true
+	}
```

Comment:
This can be simplified with 1 neo4j query, you can take example on:
https://github.com/deepfence/ThreatMapper/blob/v2/deepfence_server/controls/version.go#L163